server/db/models/campaigns.js

```diff
@@ -10,5 +11,76 @@ export default db.Model.extend({
   },
   calls() {
     return this.hasMany(Call);
+  },
+  responses() {
+    return this.hasMany(Response).through(Call);
+  },
+  getExport() {
+    // getExport should only be called on campaigns that were loaded with related calls.responses
+    const calls = this.relations.calls;
+
+    // some records have duplicate props, this renames the props and returns
+    // a version of the object that no longer has the duplicate prop name.
+    function samePropProcessing(object, objectType, propName, index) {
+      // create a copy because linter likes functional
+      const cloneObject = object;
+      if (index) {
+        // if we're given an index, it's for naming purposes, use it
+        cloneObject[`${objectType}_${index}_${propName}`] = cloneObject[propName];
+      } else {
+        cloneObject[`${objectType}_${propName}`] = cloneObject[propName];
+      }
+      // delete the potential duplicate prop name
+      delete cloneObject[propName];
+      return cloneObject;
+    }
+    // this will be an accumulator that will collect csv headers from record keys
+    let headers = [];
+    // make an array of strings that are lines in the csv.
+    const joined = calls.map((call) => {
+      const responses = call.relations.responses;
+      // this variable will hold the complete record for one line in the csv
+      // but we want to set it conditionally
+      let record;
+      // reassign the cloned call with each processing to keep it clean as we go.
+      let callClone = samePropProcessing(call.attributes, 'call', 'id');
```

Comment:
Oh ok, I see. So it looks like call.attributes is never referenced after this, so looks like we're bug free.
If it were referenced after however, have you considered how that may change expected behavior? 
I think it's a good rule of thumb not to mutate input, but in this case it's cool. Just my two cents 😄

Reply:
I agree! Thought it was handled since the linter didn't complain about reassignment, but now I know it's not, I'll get around this with a shallow clone via lodash now.